plugins/processors/awsappsignals/internal/resolver/attributesresolver.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package resolver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/zap"
+)
+
+var DefaultHostedInAttributes = map[string]string{
+	AttributeHostedInEnvironment: HostedInAttributeEnvironment,
+}
+
+type subResolver interface {
+	Process(attributes, resourceAttributes pcommon.Map) error
+	Stop(ctx context.Context) error
+}
+
+type attributesResolver struct {
+	subResolvers []subResolver
+}
+
+// create a new attributes resolver
+func NewAttributesResolver(resolverNames []string, logger *zap.Logger) *attributesResolver {
+	subResolvers := []subResolver{}
+	for _, resolverName := range resolverNames {
+		if resolverName == "eks" {
+			subResolvers = append(subResolvers, getEksResolver(logger), newEKSHostedInAttributeResolver())
+		} else {
+			subResolvers = append(subResolvers, newHostedInAttributeResolver(DefaultHostedInAttributes))
+		}
+	}
+	return &attributesResolver{
+		subResolvers: subResolvers,
+	}
+}
+
+// Process the attributes
+func (r *attributesResolver) Process(attributes, resourceAttributes pcommon.Map, _ bool) error {
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Process(attributes, resourceAttributes); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *attributesResolver) Stop(ctx context.Context) error {
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Stop(ctx); err != nil {
```

Comment:
Doesn't this mean that if the first subResolver fails to stop, none of the others will be stopped? Can we use an `errors.Join` to just combine all the errors we get from trying to stop the subResolvers and then return that?

Reply:
changed to the following

```
func (r *attributesResolver) Stop(ctx context.Context) error {
	var errs error
	for _, subResolver := range r.subResolvers {
		if err := subResolver.Stop(ctx); err != nil {
			errs = errors.Join(errs, err)
		}
	}
	return errs
}
```

Reply:
nit: Could do 

```go
func (r *attributesResolver) Stop(ctx context.Context) error {
	var errs error
	for _, subResolver := range r.subResolvers {
        errs = errors.Join(errs, subResolver.Stop(ctx))
	}
	return errs
}
```